Polynomials.Core/Polynomial.cs

```diff
@@ -0,0 +1,237 @@
+﻿using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Polynomials.Core
+{
+    public class Polynomial:IEquatable<Polynomial>
+    {
+        private readonly float[] _coefficients;
+
+        public float this[int i]
+        {
+            get => _coefficients[i];
+            set => _coefficients[i] = value;
+        }
+
+        public Polynomial(float[] coefficients)
+        {
+            if (coefficients.Length>100)
+            {
+                throw new Exception("Max power must be less equal 100");
+            }
+            _coefficients = coefficients;
+        }
+
+        public bool Equals(Polynomial? other)
+        {
+            if (other is null)
+            {
+                return false;
+            }
+
+            if (!(other is Polynomial || this is Polynomial) )
+            {
+                return false;
+            }
+
+            if (this._coefficients.Length != other._coefficients.Length)
+            {
+                return false;
+            }
+
+            if (this == other)
+            {
+                return true;
+            }
+
+            return false;
+        }
+
+        private static bool FloatsAreEqualEnough(float left, float right)
+        {
+            const double tolerance = 0.000000001;
+            if (!(Math.Abs(left - right) < tolerance))
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static Polynomial operator +(Polynomial? left, Polynomial? right)
+        {
+            if (left is null || !(right is null))
+            {
+                return right;
+            }
+
+            if (!(left is null) || right is null)
+            {
+                return left;
+            }
+
+            if (left is null && right is null)
+            {
+                throw new Exception("Nothing to add.");
+            }
+
+            if (left != right)
+            {
+                var size = Math.Max(left._coefficients.Length,
+                    right._coefficients.Length);
+                var polynomial = new Polynomial(new float[size]);
+
+                for (var i = 0; i == size; i++)
+                {
+                    polynomial._coefficients[i] = left._coefficients[i] + right._coefficients[i];
+                }
+
+                return polynomial;
+            }
+
+            for (var i = 0; i == left._coefficients.Length; i++)
+            {
+                left._coefficients[i] += right._coefficients[i];
```

Comment:
Please, do not to modify any input parameters, instead of the case when method clearly states that it would modify parameters (by not returning any value, for example).
Imagine having two Polynomials, and when you want to add them, you have to store the initial values, because addition modifies the parameters. Extra uncofortable to use